### PR TITLE
MRC-2976 Deployment - Updated travis config to use dpl V2

### PR DESCRIPTION
**Pull request by vercasson**
- [x] This change is documented in CHANGELOG.md

https://docs.travis-ci.com/user/deployment-v2

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,10 +31,9 @@ deploy:
   # Copy the files in `gcs-files/` to Google Cloud Storage. This is used for
   # static assets, namely a lot of images.
   - provider: gcs
-    cleanup: false
-    skip_cleanup: true
```
Comment (vercasson) [inline .travis.yml:36]: cleanup defaults to `false` now.

File: .travis.yml [hunk 2]
```diff
@@ -47,22 +45,28 @@ deploy:
       tags: true
   # Publish to npm
   - provider: npm
-    cleanup: false
-    skip_cleanup: true
```
Comment (vercasson) [inline .travis.yml:51]: cleanup defaults to false now.

File: .travis.yml [hunk 3]
```diff
@@ -47,22 +45,28 @@ deploy:
...
     email: $NPM_EMAIL
-    api_key: $NPM_TOKEN
+    api_token: $NPM_TOKEN
     tag: latest
     on:
       tags: true
   # Update GitHub pages for the documentation and demos
-  - provider: pages
-    cleanup: false
-    skip_cleanup: true
```
Comment (vercasson) [inline .travis.yml:60]: cleanup defaults to false now.

File: .travis.yml [hunk 4]
```diff
@@ -17,7 +17,6 @@ cache:
   npm: true
 before_install:
   # - npm i -g npm@9.8.1
```
Comment (vercasson) [inline .travis.yml:19]: Keeping this commented line until we get a few successful NPM deploys without an auth related error. Restoring this line downgrades our NPM version, but can solve the auth issue. When it failed in the past I simply deployed to NPM from my local machine where I fixed the auth issue in my npm config a while ago. Hopefully opting-in to dpl V2 will fix the problem on the travis VM.

File: .travis.yml [hunk 5]
```diff
@@ -47,22 +45,28 @@ deploy:
...
+    edge:
+      branch: release_20240617
     email: $NPM_EMAIL
-    api_key: $NPM_TOKEN
+    api_token: $NPM_TOKEN
     tag: latest
     on:
       tags: true
   # Update GitHub pages for the documentation and demos
-  - provider: pages
```
Comment (vercasson) [inline .travis.yml:58]: See below, provider is now `pages:git`

**Review by tarakreft — APPROVED** [review]

**Review by amyhliao — APPROVED** [review]

**Pull request merged**